### ISSUE TITLE
Error with ec2_group module - Invalid rule parameter '-'

### ISSUE THREAD

**Issue by harsha-y** [ansible/ansible-modules-core#5097]
##### ISSUE TYPE

- Bug Report
##### COMPONENT NAME

ec2_group
##### ANSIBLE VERSION

```
ansible 2.1.2.0
  config file = /Users/user/git/ansible/ansible.cfg
  configured module search path = Default w/o overrides
```
##### CONFIGURATION

```
[defaults]
retry_files_enabled = False
gathering = smart
nocows = 1
roles_path = /etc/ansible/roles

[privilege_escalation]

[paramiko_connection]

[ssh_connection]

[accelerate]

[selinux]

[colors]
```
##### OS / ENVIRONMENT

`macOS Sierra‎ version 10.12`
##### SUMMARY

Trying to generate rules using Jinja2 throws an error about invalid parameter.
##### STEPS TO REPRODUCE

createSG.yml

```

---
- hosts: localhost
  connection: local
  gather_facts: no

  vars:
    aws_profile_name: local_aws_profile
    aws_vpc_id: vpc-abcd1234
    ip_whitelist:
      - 8.8.8.8/32
      - 8.8.4.4/32

  tasks:
  - name: Create security group with IP blocks
    ec2_group:
      profile: "{{ aws_profile_name }}"
      region: us-east-1
      description: "Whitelist"
      name: sg-whitelist
      purge_rules: true
      rules: |
        {% for host in ip_whitelist %}
        - proto: tcp
          from_port: 443
          to_port: 443
          cidr_ip: {{ host }}
        {% endfor %}
      vpc_id: "{{ aws_vpc_id }}"
      state: present
```

```
ansible-playbook playbooks/createSG.yml
```

##### EXPECTED RESULTS

A security group created with two ingress rules. Any rules not part of the play to be purged.
##### ACTUAL RESULTS

```
PLAY [localhost] ***************************************************************

TASK [Create security group with IP blocks] *******************
fatal: [localhost]: FAILED! => {"changed": false, "failed": true, "msg": "Invalid rule parameter '-'"}

NO MORE HOSTS LEFT *************************************************************

PLAY RECAP *********************************************************************
localhost                  : ok=0    changed=0    unreachable=0    failed=1
```

**Comment by ansibot** [comment]
@ansible ping, this issue is waiting for your response.
[click here for bot help](https://github.com/ansible/ansibullbot/blob/master/ISSUE_HELP.md)

**Comment by harsha-y** [comment]
More info -

This is not a newly introduced issue. Had identical results with ansible-2.1.1.0

**Comment by ansibot** [comment]
@ansible, ping. This issue is still waiting on your response.
[click here for bot help](https://github.com/ansible/ansibullbot/blob/master/ISSUE_HELP.md)

**Comment by ansibot** [comment]
@ansible, ping. This issue is still waiting on your response.
[click here for bot help](https://github.com/ansible/ansibullbot/blob/master/ISSUE_HELP.md)

**Comment by ansibot** [comment]
@ansible, ping. This issue is still waiting on your response.
[click here for bot help](https://github.com/ansible/ansibullbot/blob/master/ISSUE_HELP.md)

**Comment by ansibot** [comment]
@ansible, ping. This issue is still waiting on your response.
[click here for bot help](https://github.com/ansible/ansibullbot/blob/master/ISSUE_HELP.md)

**Comment by ansibot** [comment]
This repository has been locked. All new issues and pull requests should be filed in https://github.com/ansible/ansible

Please read through the <a href="http://docs.ansible.com/ansible/dev_guide/repomerge.html#move-issues-and-prs-to-new-repo">repomerge</a> page in the dev guide. The guide contains links to tools which automatically move your issue or pull request to the ansible/ansible repo.

**Comment by ansibot** [comment]
This issue was migrated to https://github.com/ansible/ansible/issues/29621
